bin/improver-weighted-blending

```diff
@@ -0,0 +1,138 @@
+
+def main():
+    """Load in arguments and ensure arguments with non-None defaults are set 
+       correctly. Then load in the data to average over and calculate 
+       default weights using the method chosen before carrying out the
+       weighted average."""
+    parser = argparse.ArgumentParser(
+    description='Calculate the default weights to apply in weighted ' +
+                'blending plugins using the ChooseDefaultWeightsLinear or ' +
+                'ChooseDefaultWeightsNonLinear plugins. Then apply these ' +
+                'weights to the cube using the BasicWeightedAverage plugin.' +
+                ' Required for ChooseDefaultWeightsLinear: y0val and ONE ' +
+                'of slope, ynval. Required for ' +
+                'ChooseDefaultWeightsNonLinear: cval.')
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument('--linear', default=False, action='store_true',
+                       help='Option to use the ChooseDefaultWeightsLinear ' +
+                            'plugin.')
+    group.add_argument('--nonlinear', default=False, action='store_true',
+                       help='Option to use the ' +
+                            'ChooseDefaultWeightsNonLinear plugin.')
+
+    parser.add_argument('coordinate',
+                        metavar='COORDINATE_TO_AVERAGE_OVER',
+                        help='The coordinate over which the weighted ' +
+                             'average will be applied.')
+    parser.add_argument('input_filepath', metavar='INPUT_FILE',
+                        help='A path to an input NetCDF file to be processed.')
+    parser.add_argument('output_filepath', metavar='OUTPUT_FILE',
+                        help='The output path for the processed NetCDF.')
+    linear = parser.add_argument_group('linear weights options',
+                                       'Options for the linear weights ' +
+                                       'calculation in ' +
+                                       'ChooseDefaultWeightsLinear')
+    lin_opts = linear.add_mutually_exclusive_group()
+    lin_opts.add_argument('--slope', metavar='LINEAR_SLOPE', type=float,
+                          help='The slope of the line used for choosing ' +
+                               'default linear weights. Only one of ynval ' +
+                               'and slope may be set.')
+    lin_opts.add_argument('--ynval', metavar='LINEAR_END_POINT', 
+                          type=float,
+                          help='The relative value of the weighting end ' +
+                               'point for choosing default linear weights. ' +
+                               'Only one of ynval and slope may be set.')
+    linear.add_argument('--y0val', metavar='LINEAR_STARTING_POINT', type=float,
+                        help='The relative value of the weighting start ' +
+                             'point for choosing default linear weights.')
```

Comment:
Please can you add that if this is set then this needs to be a positive float, if y0val is not set then the calculation of weights will default to y0val=20.0 and ynval=2.0